src/activity_queue.rs

```diff
@@ -220,27 +201,145 @@ pub(crate) fn generate_request_headers(inbox_url: &Url) -> HeaderMap {
     headers
 }
 
+/// A simple activity queue which spawns tokio workers to send out requests
+/// When creating a queue, it will spawn a task per worker thread
+/// Uses an unbounded mpsc queue for communication (i.e, all messages are in memory)
+pub(crate) struct ActivityQueue {
+    // Our "background" tasks
+    senders: Vec<UnboundedSender<SendActivityTask>>,
+    // Round robin of the sender list
+    last_sender_idx: AtomicUsize,
+    // Stats shared between the queue and workers
+    stats: Arc<Stats>,
+}
+
+/// Simple stat counter to show where we're up to with sending messages
+/// This is a lock-free way to share things between tasks
+/// When reading these values it's possible (but extremely unlikely) to get stale data if a worker task is in the middle of transitioning
+#[derive(Default)]
+struct Stats {
+    pending: AtomicUsize,
+    running: AtomicUsize,
+    dead: AtomicUsize,
+    complete: AtomicUsize,
+}
```

Comment:
It seems like dead and complete would overflow if the server is running for a long time. They should be reset every hour or something similar.

Reply:
They will wrap on overflow, after `2^64-1` (i.e, after `18446744073709551615` messages), but I see the merit in resetting them periodically.  I'll adjust this accordingly.